src/dispatch.rs

```diff
@@ -239,3 +252,35 @@ enum PutError {
     OpenTransactionError(OpenTransactionError),
     CommitError(db::CommitError),
 }
+
+pub async fn init_client_id(s: &dyn Store) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
```

Comment:
We don't want this to be public in general, right, it's only for testing?

I wish I knew better what the cost of expanding the interface is, but I feel uncomfortable with it at first blush. Did you try writing the tests below without putting them in a `mod test` so they were black box?

Reply:
Good point